modules/schematics/src/reducer/files/__name@dasherize@if-flat__/__name@dasherize__.reducer.ts.template

```diff
@@ -11,11 +12,21 @@ export const initialState: State = {
 
 };
 
-export const reducer = createReducer(
-  initialState,
 <% if(feature) { %>
-  on(<%= classify(name) %>Actions.<%= prefix %><%= classify(name) %>s, state => state),
-<% if(api) { %>  on(<%= classify(name) %>Actions.<%= prefix %><%= classify(name) %>sSuccess, (state, action) => state),
-  on(<%= classify(name) %>Actions.<%= prefix %><%= classify(name) %>sFailure, (state, action) => state),
-<% } %><% } %>
-);
+export const <%= camelize(name) %>Feature = createFeature({
+  name: <%= camelize(name) %>FeatureKey,
+  reducer: <% } else {%>export const reducer = <% } %>createReducer(
+    initialState,
+    on(<%= classify(name) %>Actions.<%= prefix %><%= classify(name) %>s, state => state),
+  <% if(api) { %>  on(<%= classify(name) %>Actions.<%= prefix %><%= classify(name) %>sSuccess, (state, action) => state),
+    on(<%= classify(name) %>Actions.<%= prefix %><%= classify(name) %>sFailure, (state, action) => state),
+  <% } %>
+  )<% if(feature) { %>,
+});<% } else {%>;<% } %>
+
+<% if(feature) { %>
+  export const {
+    name,
+    reducer,
+  } = feature;
+<% } %>
```

Comment:
Same here, we can keep reducer creation as it is, so `createFeature` usage will look like this:

```
<% if(feature) { %>
  export const <%= camelize(name) %>Feature = createFeature({
    name: <%= camelize(name) %>FeatureKey,
    reducer,
  });
<% } %>
```